docs/design/node_taint_eviction.md

```diff
@@ -0,0 +1,94 @@
+This information could be consumed by the operator to proactively replace Pods that are running on nodes with a given set of taints.
+
+## General Design Goals
+
+The goal of this design is to implement a mechanism that allows the operator to replace Pods that are running on nodes with a given set of taints.
+It's expected that the operator will detect the Pods running on tainted nodes in a timely manner, either by receiving an event triggering the reconciliation loop or when running the periodic reconciliation, which will happen every 10h per default.
+The actual replacement of all Pods running on tainted nodes, depends on the data that must be replicated and the number of affected Pods.
+
+## Current Implementation
+
+We don't have a current implementation of this and the operator is currently not able to access any node information.
+The closest implementation to this is the [Replace Failed ProcessGroups reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/replace_failed_process_groups.go), which could be extended for this.
+
+## Proposed Design
+
+We have to make sure that this feature can be enabled and disabled, e.g. if someone wants to run the operator inside a cluster without node access.
+The `AutomaticReplacementOptions` will be extended to support the configuration of replacements for Pods that run on tainted nodes.
+The new field will be called `taints` and will contain a list, the default would be an empty list, of taint configurations.
+That configuration will include the taint key, this has to be an exact match or a wildcard `*`, and the duration this taint must be present on the node.
+The `durationInSeconds` allow to specify how sensitive the operator should be to taints.
+If a wildcard is specified as key this will be used as default value, except there is an exact match defined for a taint key.
+
+```yaml
+...
+automaticReplacementOptions:
+  enabled: true
+  failureDetectionTimeSeconds: 7200
+  maxConcurrentReplacements: 1
+  enableNodeFailureDetection: true
+  taints:
+  - key: "example.org/maintenance"
+    durationInSeconds: 7200 # Ensure the taint is present for at least 2 hours before replacing Pods on a node with this taint
+  - key: "*" # The wildcard would allow to define a catch all configuration
+    durationInSeconds: 3600 # Ensure the taint is present for at least 1 hour before replacing Pods on a node with this taint
+```
+
+We will add a new [ProcessGroupConditionType](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/api/v1beta2/foundationdbcluster_types.go#L673), called `RunningOnTaintedNode` to mark process groups that are running on a tainted node.
+In addition we will extend the  [conditions that need replacement](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v1.14.0/api/v1beta2/foundationdbcluster_types.go#L65) by the newly created `RunningOnTaintedNode` condition.
```

Comment:
We discussed this offline and we will actually add 2 status, one indicating that a process group has a Pod running on a tainted node and another one to indicate that a replacement can be triggered.